src/lib.rs

```diff
@@ -242,6 +251,31 @@ impl<T: ContactInfo> RoutingTable<T> {
         }
     }
 
+    /// Removes `name` from routing table and returns `false` if we no longer need to stay connected.
+    ///
+    /// We should remain connected iff entry at bucket index of `name` is in the routing table and is
+    /// within the `GROUP_SIZE` closest nodes in that bucket.
+    pub fn need_to_keep(&mut self, name: &XorName) -> bool {
+        if name == self.our_name() {
+            return true;
+        }
+
+        let not_needed = self.buckets[self.bucket_index(name)]
+                             .iter()
+                             .skip(GROUP_SIZE)
+                             .cloned()
+                             .collect::<Vec<T>>();
+
+        for can_remove in &not_needed {
+            if can_remove.name() == name {
+                let _ = self.remove(name);
+                return false
+            }
+        }
```

Comment:
This method could be simplified a lot: Just check whether `self.search(name)` is `(_, Ok(i))` with `i >= GROUP_SIZE`. In that case, `self.remove(name)` and return `true`, otherwise return `false`.